Shared/Samples/Search for web map/SearchForWebMapView.swift

```diff
@@ -0,0 +1,189 @@
+                            if item.id == model.portalItems.last?.id {
+                                model.findNextItems()
+                            }
+                        }
+                    }
+                }
+                
+                ProgressView()
+                    .padding()
+                    .opacity(resultsAreLoading ? 1 : 0)
+            }
+            
+            VStack {
+                Text("No Results")
+                    .font(.headline)
+                Text("Check spelling or try a new search.")
+                    .font(.footnote)
+            }
+            .opacity(!resultsAreLoading && !query.isEmpty && model.portalItems.isEmpty ? 1 : 0)
+        }
+        .background(Color(.secondarySystemBackground))
+        .searchable(
+            text: $query,
+            placement: .navigationBarDrawer(displayMode: .always),
+            prompt: "Search for a web map"
+        )
+        .onChange(of: query) { _ in
+            // Load new results when the query changes.
+            model.findItems(for: query)
+        }
+        .task(id: model.task) {
+            guard model.task != nil else { return }
+            
+            resultsAreLoading = true
+            defer { resultsAreLoading = false }
+            
+            do {
+                try await model.task?.value
+            } catch {
+                self.error = error
+            }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension SearchForWebMapView {
+    /// A map view that shows an alert and dismisses itself when there is an error loading the map.
+    struct SafeMapView: View {
+        /// The map to show in the map view.
+        let map: Map
+        
+        /// The action to dismiss the view.
+        @Environment(\.dismiss) private var dismiss: DismissAction
+        
+        /// A Boolean value indicating whether the map is being loaded.
+        @State private var mapIsLoading = false
+        
+        /// The error shown in the error alert.
+        @State private var error: Error?
+        
+        var body: some View {
+            ZStack {
+                MapView(map: map)
+                    .task {
+                        mapIsLoading = true
+                        defer { mapIsLoading = false }
+                        
+                        // Show an alert for an error loading the map.
+                        do {
+                            try await map.load()
+                        } catch {
+                            self.error = error
+                        }
+                    }
+                    .errorAlert(presentingError: $error)
+                    .onChange(of: error == nil) { _ in
+                        // Dismiss the view once the error alert is dismissed.
+                        guard error == nil else { return }
+                        dismiss()
+                    }
```

Comment:
If we call dismiss on `SafeMapView`, then both this view and the alert will be dismissed, as the alert is a subview to this view.

I don't think the auto-dismiss is too important. Also I think it is fine to not use `errorAlert`, but to implement a separate one for this view, so we have control over the done action.